diff --git a/dist/index.mjs b/dist/index.mjs
index 749b5d65546011b9d87d0a28dd86c1de091efe47..c6630f190eb371a5727f1b37325e848f96c02bd0 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -722,6 +722,9 @@ const createArgosLegacyAPIClient = (options)=>{
             });
             return response.data;
         } catch (error) {
+            console.error(error.response);
+            console.error(error.response.details);
+            console.error(JSON.stringify(error.response.details, null, 2));
             if (error?.response?.data?.error?.message) {
                 // @ts-ignore
                 throw new Error(error.response.data.error.message, {
